cf-check/dump.c

```diff
@@ -266,72 +251,100 @@ void nice_print_persistent_class(const MDB_val value)
+    if (structs)
     {
-        if (StringSafeEqual(key.mv_data, "DATABASE_AGE"))
+        if (StringEndsWith(file, "cf_lastseen.lmdb")
+            && StringStartsWith(key.mv_data, "q"))
+        {
+            print_lastseen_quality(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_lock.lmdb"))
+        {
+            print_lock_data(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_observations.lmdb"))
+        {
+            if (StringSafeEqual(key.mv_data, "DATABASE_AGE"))
+            {
+                assert(sizeof(double) == value.mv_size);
+                const double *const age = value.mv_data;
+                printf("%f", *age);
+            }
+            else
+            {
+                print_averages(value, strip_strings);
+            }
+        }
+        else if (StringEndsWith(file, "history.lmdb"))
         {
-            assert(sizeof(double) == value.mv_size);
-            const double *const age = value.mv_data;
-            printf("%f", *age);
+            print_averages(value, strip_strings);
+        }
+        else if (StringEndsWith(file, "cf_state.lmdb"))
+        {
+            print_persistent_class(value, strip_strings);
         }
         else
         {
-            nice_print_averages(value);
+            fallback = true;
         }
     }
-    else if (StringEndsWith(file, "history.lmdb"))
-    {
-        nice_print_averages(value);
-    }
-    else if (StringEndsWith(file, "cf_state.lmdb"))
-    {
-        nice_print_persistent_class(value);
-    }
-    else
+
+    const bool was_printed = (structs && !fallback);
+    if (!was_printed)
     {
-        dump_print_json_string_nice(value.mv_data, value.mv_size);
+        dump_print_json_string(value.mv_data, value.mv_size, strip_strings);
     }
+}
+
+void dump_print_object_line(
+    const MDB_val key,
+    const MDB_val value,
+    const char *const file,
+    const bool strip_strings,
+    const bool structs)
+{
+    printf("\t");
+    dump_print_json_string(key.mv_data, key.mv_size, strip_strings);
+    printf(": ");
+    dump_print_value(key, value, file, strip_strings, structs);
     printf(",\n");
 }
 
-void dump_print_array_line(const MDB_val value)
+void dump_print_array_line(const MDB_val value, const bool strip_strings)
 {
     printf("\t");
-    dump_print_json_string(value.mv_data, value.mv_size);
+    dump_print_json_string(value.mv_data, value.mv_size, strip_strings);
     printf(",\n");
 }
 
 void dump_print_closing_bracket(const dump_mode mode)
 {
-    if (mode == DUMP_MODE_SIMPLE || mode == DUMP_MODE_NICE)
+    if (mode == DUMP_MODE_KEYS || mode == DUMP_MODE_VALUES)
     {
-        printf("}\n");
+        printf("]\n");
     }
     else
     {
-        printf("]\n");
+        printf("}\n");
     }
 }
 
 int dump_db(const char *file, const dump_mode mode)
 {
     assert(file != NULL);
 
+    const bool strip_strings = ((mode == DUMP_MODE_NICE) ? true : false);
```

Comment:
Why not just `strip_strings = (mode == DUMP_MODE_NICE)`?

Reply:
Haha, too much copy pasting and refactoring. Fixed, thanks!